sdk/nodejs/automation/workspace.ts

```diff
@@ -229,10 +230,12 @@ export interface Workspace {
      */
     removeStack(stackName: string): Promise<void>;
     /**
-     * Returns all Stacks created under the current Project.
-     * This queries underlying backend and may return stacks not present in the Workspace (as Pulumi.<stack>.yaml files).
+     * Returns all Stacks from the underlying backend based on the provided options.
+     * This queries backend and may return stacks not present in the Workspace (as Pulumi.<stack>.yaml files).
+     *
+     * @param opts Options to customize the behavior of the list.
      */
-    listStacks(): Promise<StackSummary[]>;
+    listStacks(opts: ListOptions): Promise<StackSummary[]>;
```

Comment:
```suggestion
    listStacks(opts?: ListOptions): Promise<StackSummary[]>;
```

I think this is necessary to make it backwards compatible.

Or maybe it isn't given https://github.com/pulumi/pulumi/pull/16266/files#diff-6639d4f3b8e965460d136477f5e08ed87621356d7747fd0c50b1cba420704e06R170 works :thinking: ?

Reply:
ah, good catch. I intended for those to be equivalent. Apparently it works because the implementation can have an optional version of the param as long as the object type of ListOptions matches. TIL.